studiopress/genesis-custom-blocks
# Fix bug with same field name, add more e2e tests

#### Changes
* Bump version from `1.1.0-beta-1` to `1.1.0`, for a stable release that might happen this week
* Allow 2 fields to look like they have the same name 
* Add a Default setting to the Email field
* Remove `npm run test:js` from pre-commit hook, as it was taking too long. CircleCI still runs it.
* Change 'Field' to 'Field Type': <img width="839" alt="thi-bsdf" src="https://user-images.githubusercontent.com/4063887/104647299-44c83300-5677-11eb-88ec-7b6f5c5433a7.png">
* Add much better e2e coverage. Every field is now tested, including its front-end output.

#### Testing instructions
1. `composer install && npm install && npm run dev`
1. `/wp-admin` > Custom Blocks > Add New
1. Add a field
1. Enter 'Hero' as the 'Field Label': 
![hero-field-label](https://user-images.githubusercontent.com/4063887/105671156-dc175c80-5ea7-11eb-8d2c-a7c3a7cd4556.png)
1. Add another field
1. Expected: it's possible to set its 'Field Label' to 'Hero Image': ![hero-image](https://user-images.githubusercontent.com/4063887/105671406-45976b00-5ea8-11eb-9f78-988f2fb3f135.png)
